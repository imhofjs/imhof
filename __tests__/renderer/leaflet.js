import tileConfigs from '../../src/tiles';
import mockMapElement from './mocks/map-element';

jest.doMock('../../src/renderer/leaflet/map-element', () => mockMapElement);

const { default: renderLeaflet } = require('../../src/renderer/leaflet');

let config;

beforeEach(() => {
  document.body.innerHTML = '<div id="leaflet-container"></div>';

  config = {
    dpi: 150,
    view: {
      center: [1, 2],
      zoom: 3,
    },
  };
});

function fakeLoadAllImages() {
  [...document.getElementsByTagName('img')].forEach((image) => {
    image.dispatchEvent(new Event('load'));
  });
}

test('renderLeaflet', async () => {
  setImmediate(fakeLoadAllImages);
  await renderLeaflet(tileConfigs.OSM, config);

  expect(document.body).toMatchSnapshot();
});
