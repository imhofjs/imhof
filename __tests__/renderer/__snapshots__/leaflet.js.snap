// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderLeaflet 1`] = `
<body>
  <div
    id="leaflet-container"
    style="display: block; transform: scale(0.64);"
  >
    <div
      class="leaflet-container leaflet-grab leaflet-touch-drag"
      style="width: 156.25%; height: 156.25%; position: relative;"
      tabindex="0"
    >
      <div
        class="leaflet-pane leaflet-map-pane"
        style="left: 0px; top: 0px;"
      >
        <div
          class="leaflet-pane leaflet-tile-pane"
        >
          <div
            class="leaflet-layer "
            style="z-index: 1;"
          >
            <div
              class="leaflet-tile-container leaflet-zoom-animated"
              style="z-index: Infinity; left: 0px; top: 0px;"
            >
              <img
                alt=""
                class="leaflet-tile leaflet-tile-loaded"
                role="presentation"
                src="https://a.tile.openstreetmap.org/3/3/3.png"
                style="width: 256px; height: 256px; left: -211px; top: -169px;"
              />
              <img
                alt=""
                class="leaflet-tile leaflet-tile-loaded"
                role="presentation"
                src="https://b.tile.openstreetmap.org/3/4/3.png"
                style="width: 256px; height: 256px; left: 45px; top: -169px;"
              />
              <img
                alt=""
                class="leaflet-tile leaflet-tile-loaded"
                role="presentation"
                src="https://b.tile.openstreetmap.org/3/3/4.png"
                style="width: 256px; height: 256px; left: -211px; top: 87px;"
              />
              <img
                alt=""
                class="leaflet-tile leaflet-tile-loaded"
                role="presentation"
                src="https://c.tile.openstreetmap.org/3/4/4.png"
                style="width: 256px; height: 256px; left: 45px; top: 87px;"
              />
            </div>
          </div>
        </div>
        <div
          class="leaflet-pane leaflet-shadow-pane"
        />
        <div
          class="leaflet-pane leaflet-overlay-pane"
        />
        <div
          class="leaflet-pane leaflet-marker-pane"
        />
        <div
          class="leaflet-pane leaflet-tooltip-pane"
        />
        <div
          class="leaflet-pane leaflet-popup-pane"
        />
      </div>
      <div
        class="leaflet-control-container"
      >
        <div
          class="leaflet-top leaflet-left"
        />
        <div
          class="leaflet-top leaflet-right"
        />
        <div
          class="leaflet-bottom leaflet-left"
        />
        <div
          class="leaflet-bottom leaflet-right"
        />
      </div>
    </div>
  </div>
</body>
`;
